pkg/kritis/review/validating_transport.go

```
@@ -101,51 +103,74 @@ func (avt *AttestorValidatingTransport) validatePublicKeyId(pubKey v1beta1.Publi
 			continue
 		}
-		if _, ok := keys[pubKey.KeyId]; ok {
-			glog.Warningf("Overwriting key with same fingerprint %s for %q.", pubKey.KeyId, avt.Attestor.Name)
+		// Base64 decode the attestor's key
+		decodedKey, err := base64.StdEncoding.DecodeString(attestorKey.AsciiArmoredPgpPublicKey)
+		if err != nil {
+			glog.Infof("Cannot base64 decode public key: %v", err)
+			continue
 		}
-		keys[pubKey.KeyId] = pubKey
+		publicKey := cryptolib.NewPublicKey(cryptolib.Pgp, decodedKey, attestorKey.KeyId)
+		publicKeys = append(publicKeys, publicKey)
 	}
-	if len(keys) == 0 {
+	if len(publicKeys) == 0 {
 		return nil, fmt.Errorf("unable to find any valid key for %q", avt.Attestor.Name)
 	}
 
-	out := []attestation.ValidatedAttestation{}
-	host, err := container.NewAtomicContainerSig(image, map[string]string{})
+	// Check for properly formatted image digest
+	digest, err := name.NewDigest(image, name.StrictValidation)
 	if err != nil {
-		glog.Error(err)
-		return nil, err
+		return nil, errors.Wrap(err, "error extracting image digest")
 	}
-	rawAtts, err := avt.Client.Attestations(image, &avt.Attestor)
+
+	verifier, err := cryptolib.NewVerifier(digest.DigestStr(), publicKeys)
 	if err != nil {
-		glog.Error(err)
-		return nil, err
+		return nil, errors.Wrap(err, "error creating verifier")
 	}
 
+	rawAtts, err := avt.Client.Attestations(image, &avt.Attestor)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching attestations for image %s: %v", image, err)
+	}
 	for _, rawAtt := range rawAtts {
-		if rawAtt.SignatureType != metadata.PgpSignatureType {
-			return nil, fmt.Errorf("Signature type %s is not supported for Attestation %v", rawAtt.SignatureType.String(), rawAtt)
+		if rawAtt.SignatureType != metadata.PgpSignatureType && rawAtt.SignatureType != metadata.GenericSignatureType {
+			glog.Warningf("Skipping attestation with unsupported signature type %s", rawAtt.SignatureType.String())
+			continue
```

To silently pass unsupported attestation type can potentially makes it hard to debug. That was the reason for failing it previously
If we intend to change this behavior,  it probably should be done in a separate PR.

---

Even if 1 Attestation has an unsupported type, wouldn't you still want to check the other Attestations retrieved by the client?

---

If we are changing behavior of kritis, we should probably do it in a separate PR.

---

Done. Reverted